### ISSUE TITLE
Downloaded library crashes when too big

### ISSUE THREAD

**Issue by matou008** [fast4x/RiMusic#1724]
### Steps to reproduce the bug

1. Open the app with >1000 songs downloaded
2. Go to Library
3. Click on Downloaded

### Expected behavior

Should list the downloaded songs

### Actual behavior

App crashes

### Logs

```
05-01 14:28:05.092  1547  1575 I ActivityTaskManager: Displayed it.fast4x.rimusic/.MainActivity: +1s40ms
05-01 14:28:05.137   727   727 I hwcomposer: [HWCDisplay] [Display_0 (type:1)] fps:60.170334,dur:1013.79,max:18.38,min:14.63
05-01 14:28:05.346  3623  4316 D WireGuard/GoBackend/ProtonTunnel: peer(xw17…SUnA) - Sending keepalive packet
05-01 14:28:05.352  6026  6316 D WireGuard/GoBackend/ProtonTunnel: peer(szjI…/0w0) - Receiving keepalive packet
05-01 14:28:05.391   365   365 I hwservicemanager: getTransport: Cannot find entry android.hardware.graphics.allocator@3.0::IAllocator/default in either framework or device manifest.
05-01 14:28:05.392 15337 15360 W Gralloc3: allocator 3.x is not supported
05-01 14:28:06.149   727   727 I hwcomposer: [HWCDisplay] [Display_0 (type:1)] fps:57.293667,dur:1012.33,max:50.16,min:14.68
05-01 14:28:06.460   889  1099 E Situation: processEvent, handle:84, flush_action:0
05-01 14:28:06.871  1547  1747 E WifiVendorHal: getWifiLinkLayerStats_1_3_Internal(l.927) failed {.code = ERROR_NOT_SUPPORTED, .description = }
05-01 14:28:07.018   889  1099 E Situation: processEvent, handle:84, flush_action:0
05-01 14:28:07.056   743   806 I libPowerHal: 8: cpu_ctrl set freq: -1 -1 -1 -1
05-01 14:28:07.165   727   727 I hwcomposer: [HWCDisplay] [Display_0 (type:1)] fps:3.938910,dur:1015.51,max:948.67,min:16.24
05-01 14:28:07.508   889  1099 E Situation: processEvent, handle:84, flush_action:0
05-01 14:28:08.179   727   727 I hwcomposer: [HWCDisplay] [Display_0 (type:1)] fps:3.943050,dur:1014.44,max:947.71,min:16.35
05-01 14:28:08.677   889  1099 E Situation: processEvent, handle:86, flush_action:0
05-01 14:28:09.194   727   727 I hwcomposer: [HWCDisplay] [Display_0 (type:1)] fps:5.914842,dur:1014.40,max:517.39,min:15.05
05-01 14:28:09.898  1547  1747 E WifiVendorHal: getWifiLinkLayerStats_1_3_Internal(l.927) failed {.code = ERROR_NOT_SUPPORTED, .description = }
05-01 14:28:10.224   727   727 I hwcomposer: [HWCDisplay] [Display_0 (type:1)] fps:3.880982,dur:1030.67,max:947.56,min:16.34
05-01 14:28:10.475   889  1099 E Situation: processEvent, handle:84, flush_action:0
05-01 14:28:10.696   743   806 I libPowerHal: 2: cpu_ctrl set freq: 1508000 -1 1417000 -1
05-01 14:28:11.180 15337 15452 E SQLiteLog: (1) too many SQL variables
05-01 14:28:11.238   727   727 I hwcomposer: [HWCDisplay] [Display_0 (type:1)] fps:4.932775,dur:1013.63,max:918.40,min:13.17
05-01 14:28:11.338 15337 15337 E AndroidRuntime: FATAL EXCEPTION: main
05-01 14:28:11.338 15337 15337 E AndroidRuntime: Process: it.fast4x.rimusic, PID: 15337
05-01 14:28:11.338 15337 15337 E AndroidRuntime: android.database.sqlite.SQLiteException: too many SQL variables (code 1 SQLITE_ERROR): , while compiling: SELECT * FROM Song WHERE id in (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:986)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:593)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:61)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1443)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1418)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at J2.b.X(SourceFile:26)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at E2.y.k(SourceFile:35)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at B6.e.L0(SourceFile:12)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at U4.q.a(SourceFile:386)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at U4.q.call(SourceFile:195)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at E2.b.t(SourceFile:81)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at C5.a.i(SourceFile:9)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at T5.J.run(SourceFile:95)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at g.J.run(SourceFile:102)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:919)
05-01 14:28:11.338 15337 15337 E AndroidRuntime:        Suppressed: Y5.h: [androidx.compose.ui.platform.D0@276ad09, J.g@3369b0e, v0{Cancelling}@c52792f, c0@9641b3c]
05-01 14:28:11.342  1547 15461 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
05-01 14:28:11.342  1547  8898 W ActivityTaskManager:   Force finishing activity it.fast4x.rimusic/.MainActivity
05-01 14:28:11.342  1547  8898 D WindowManager: display 0 setAppTransitionState 0
05-01 14:28:11.344  1547  8898 V WindowManager: Changing focus from Window{c66a2f u0 it.fast4x.rimusic/it.fast4x.rimusic.MainActivity} to null displayId=0
05-01 14:28:11.345  1547  8898 D WindowManager: display 0 setAppTransitionState 0
05-01 14:28:11.355  1547  1576 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
05-01 14:28:11.355 15337 15337 I Process : Sending signal. PID: 15337 SIG: 9
05-01 14:28:11.363  1547  1576 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
05-01 14:28:11.401  1547  1708 W InputDispatcher: channel 'c66a2f it.fast4x.rimusic/it.fast4x.rimusic.MainActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
05-01 14:28:11.401  1547  1708 E InputDispatcher: channel 'c66a2f it.fast4x.rimusic/it.fast4x.rimusic.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
05-01 14:28:11.402  1547  8883 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ LISTEN id=12603, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&NOT_VPN&FOREGROUND Uid: 10231] ], android.os.BinderProxy@56a2f6)
05-01 14:28:11.402  1547  8898 I WindowManager: WIN DEATH: Window{c66a2f u0 it.fast4x.rimusic/it.fast4x.rimusic.MainActivity}
05-01 14:28:11.402  1547  8898 W InputDispatcher: Attempted to unregister already unregistered input channel 'c66a2f it.fast4x.rimusic/it.fast4x.rimusic.MainActivity (server)'
05-01 14:28:11.402  1547  3013 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ TRACK_DEFAULT id=12602, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10231] ], android.os.BinderProxy@ace86f7)
05-01 14:28:11.402  1547  7226 D MediaSessionService: Media button session is changed to null
05-01 14:28:11.402   754 18647 D AudioPolicyService: removeNotificationClient mNotificationClients.size() 10
05-01 14:28:11.402   754  7506 D AudioFlinger: AudioFlinger::NotificationClient::binderDied pid: 15337
05-01 14:28:11.402   754  7506 D AudioFlinger: AudioFlinger::NotificationClient::~NotificationClient() pid: 15337
05-01 14:28:11.405   703   703 I Zygote  : Process 15337 exited due to signal 9 (Killed)
05-01 14:28:11.405  1547  1645 I ActivityManager: Process it.fast4x.rimusic (pid 15337) has died: vis+99 TOP
05-01 14:28:11.405  1547  1578 I libprocessgroup: Successfully killed process cgroup uid 10231 pid 15337 in 0ms
05-01 14:28:11.406   889   889 I Accelerometer: batch: handle:0, flag:0,samplingPeriodNs:20000000 maxBatchReportLatencyNs:2147483647000
05-01 14:28:11.407  1547  1793 D ConnectivityService: releasing NetworkRequest [ TRACK_DEFAULT id=12602, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10231] ] (release request)
05-01 14:28:11.409  1547  1645 D ActivityManager: makeInactive, ProcessRecord{1bf4b37 15337:it.fast4x.rimusic/u0a231}, clear holder.state
05-01 14:28:11.409  1547  1645 W ActivityManager: Scheduling restart of crashed service it.fast4x.rimusic/.service.PlayerService in 1000ms
05-01 14:28:11.421  1547  1645 I WindowManager: Window drawn AppWindowToken{3336262 token=Token{fdf6d2d ActivityRecord{45b0844 u0 it.fast4x.rimusic/.MainActivity t3959}}}
05-01 14:28:11.433  1547  1645 D WindowManager: display 0 setAppTransitionState 0
05-01 14:28:11.437  1547  1645 D WindowManager: display 0 setAppTransitionState 1
05-01 14:28:11.457  1547  1569 W ActivityManager: setHasOverlayUi called on unknown pid: 15337
05-01 14:28:11.465  1547  8883 V WindowManager: Changing focus from null to Window{bfc12e6 u0 fr.neamar.kiss/fr.neamar.kiss.MainActivity} displayId=0
05-01 14:28:11.469 25891 25925 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
05-01 14:28:11.486  1547  1570 I WindowManager:   SURFACE show Surface(name=com.android.systemui.ImageWallpaper)/@0x747493e on display:0: com.android.systemui.ImageWallpaper
05-01 14:28:11.489  1547  1570 I WindowManager: Window drawn AppWindowToken{d862768 token=Token{2b9aa8b ActivityRecord{7b8885a u0 fr.neamar.kiss/.MainActivity t3929}}}
05-01 14:28:11.490  1547  1570 D WindowManager: display 0 setAppTransitionState 2
05-01 14:28:11.490  1547  1570 D WindowManager: display 0 setAppTransitionState 0
05-01 14:28:11.495  1547  1570 I WindowManager:   SURFACE show Surface(name=fr.neamar.kiss/fr.neamar.kiss.MainActivity)/@0x89b1ace on display:0: fr.neamar.kiss/fr.neamar.kiss.MainActivity
05-01 14:28:11.499  9017  9017 I LatinIME: onFinishInput
05-01 14:28:11.845  1547  1569 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{45b0844 u0 it.fast4x.rimusic/.MainActivity t-1 f}
05-01 14:28:12.319 32709 15468 E memtrack: Couldn't load memtrack module
05-01 14:28:12.319 32709 15468 W android.os.Debug: failed to get memory consumption info: -1
05-01 14:28:12.423   743   806 I libPowerHal: 2: cpu_ctrl set freq: -1 -1 -1 -1
05-01 14:28:12.434   703   703 D Zygote  : Forked child process 15469
05-01 14:28:12.438  1547  1577 I ActivityManager: Start proc 15469:it.fast4x.rimusic/u0a231 for service {it.fast4x.rimusic/it.fast4x.rimusic.service.PlayerService}
05-01 14:28:12.449 15469 15469 E .fast4x.rimusi: Not starting debugger since process cannot load the jdwp agent.
05-01 14:28:12.536 15469 15469 W .fast4x.rimusi: Accessing hidden field Ljava/util/Collections$SynchronizedCollection;->mutex:Ljava/lang/Object; (greylist-max-o, reflection, denied)
05-01 14:28:12.537 15469 15469 W .fast4x.rimusi: Accessing hidden method Ljava/util/Collections$SynchronizedSet;-><init>(Ljava/util/Set;Ljava/lang/Object;)V (greylist-max-o, reflection, denied)
05-01 14:28:12.537 15469 15469 W .fast4x.rimusi: Accessing hidden method Ljava/util/Collections$SynchronizedCollection;-><init>(Ljava/util/Collection;Ljava/lang/Object;)V (greylist-max-o, reflection, denied)
05-01 14:28:12.607 15469 15469 D NetworkSecurityConfig: No Network Security Config specified, using platform default
05-01 14:28:12.623 15469 15469 I ExoPlayerImpl: Init d61b5e5 [AndroidXMedia3/1.3.0] [lima, motorola one macro, motorola, 29]
05-01 14:28:12.660 15469 15469 W MediaSessionCompat: Couldn't find a unique registered media button receiver in the given context.
05-01 14:28:12.662  1547  3728 D MediaSessionService: Media button session is changed to it.fast4x.rimusic/PlayerService (userId=0)
```

### RiMusic version

0.6.31

### Android version

Android 10

### Additional information

Very similar to closed [issue 1386](https://github.com/fast4x/RiMusic/issues/1386)

**Comment by aneesh1122** [comment]
off-topic but how do you obtain log?

**Comment by fast4x** [comment]
Fixed, available from version 0.6.33

**Comment by matou008** [comment]
> off-topic but how do you obtain log?

@aneesh1122 Connect your device to a computer, enable USB debugging and use `adb logcat` (more info here https://developer.android.com/tools/logcat)


Thank you @fast4x for your fast fix and your amazing work overall
